okta/resource_okta_app_group_assignments.go

```diff
@@ -0,0 +1,305 @@
+	}
+
+	err = d.Set("group", tfFlattenedAssignments)
+	if err != nil {
+		return diag.Errorf("failed to set groups in tf state: %v", err)
+	}
+	return nil
+}
+
+func resourceAppGroupAssignmentsDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := getOktaClientFromMetadata(m)
+
+	for _, rawGroup := range d.Get("group").(*schema.Set).List() {
+		group := rawGroup.(map[string]interface{})
+
+		_, err := client.Application.DeleteApplicationGroupAssignment(
+			ctx,
+			d.Get("app_id").(string),
+			group["id"].(string),
+		)
+		if err != nil {
+			return diag.Errorf("failted to delete application group assignment: %v", err)
+		}
+	}
+	return nil
+}
+
+func resourceAppGroupAssignmentsUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := getOktaClientFromMetadata(m)
+	appID := d.Get("app_id").(string)
+
+	old, new := d.GetChange("group")
+	oldSet := old.(*schema.Set)
+	newSet := new.(*schema.Set)
+
+	toAdd := tfGroupsToGroupAssignments(
+		newSet.Difference(oldSet).List()...,
+	)
+	toRemove := tfGroupsToGroupAssignments(
+		oldSet.Difference(newSet).List()...,
+	)
+
+	err := deleteGroupAssignments(
+		client.Application.DeleteApplicationGroupAssignment,
+		ctx,
+		appID,
+		toRemove,
+	)
+	if err != nil {
+		return diag.Errorf("failed to delete group assignment: %v", err)
+	}
+
+	err = addGroupAssignments(
+		client.Application.CreateApplicationGroupAssignment,
+		ctx,
+		appID,
+		toAdd,
+	)
+	if err != nil {
+		return diag.Errorf("failed to add group assignment: %v", err)
+	}
+	return resourceAppGroupAssignmentsRead(ctx, d, m)
+}
+
+// groupAssignmentToTFGroup
+func groupAssignmentToTFGroup(assignment *okta.ApplicationGroupAssignment) (map[string]interface{}, error) {
+	profile := "{}"
+
+	jsonProfile, err := json.Marshal(assignment.Profile)
+	if err != nil {
+		return nil, err
+	}
+	if string(jsonProfile) != "" {
+		profile = string(jsonProfile)
+	}
+
+	tfAssignment := map[string]interface{}{
+		"id":       assignment.Id,
+		"priority": assignment.Priority,
+		"profile":  profile,
+	}
+	return tfAssignment, nil
+}
+
+func tfGroupsToGroupAssignments(groups ...interface{}) map[string]okta.ApplicationGroupAssignment {
+	assignments := map[string]okta.ApplicationGroupAssignment{}
+	// run through all groups in the set and create an assignment
+	for _, untypedGroup := range groups {
+		group := untypedGroup.(map[string]interface{})
+
+		id := group["id"].(string)
+		// skip empty groups with no id
+		if id == "" {
+			continue
+		}
+		priority := group["priority"].(int)
+
+		rawProfile := group["profile"]
+		var profile interface{}
+		_ = json.Unmarshal([]byte(rawProfile.(string)), &profile)
+
+		assignments[id] = okta.ApplicationGroupAssignment{
+			Profile:  profile,
+			Priority: int64(priority),
+			Id:       id,
+		}
+	}
+	return assignments
+}
+
+// paginate listAppGroupAssignments
+func listAppGroupAssignments(
```

Comment:
Please use `listApplicationGroupAssignments` func here. No need to duplicate the code :)